### PR TITLE
Fix extension combination score calculation for project creation

### LINKED ISSUE #525: stage.code.quarkus.io returning 500 in a very specific case
**Describe the bug**

Quarkus Tools for IntelliJ has an it test that selects all the extensions. It is now failing with 500 status code.

**Expected behavior**

200 should be returned

**Actual behavior**

500 status code is returned.

**To Reproduce**
1. Clone redhat-developer/intellij-quarkus
2. run `./gradlew test --tests com.redhat.devtools.intellij.quarkus.module.QuarkusModelRegistryTest`

**Screenshots**

image

**Additional context**

Previous jobs that were ok before: https://github.com/redhat-developer/intellij-quarkus/actions/workflows/ci.yml

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes https://github.com/quarkusio/code.quarkus.io/issues/525

### CODE REVIEW

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java [hunk 1]
```diff
@@ -62,7 +61,7 @@ private double calculateScore(OriginCombination s) {
         for (OriginWithPreference o : s.getCollectedOrigins()) {
             combinationScore += Math.pow(extOrigins.size(),
                     highestRegistryPreference + 1 - o.getPreference().registryPreference)
-                    * ((double) (Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
+                    * (((double) (Integer.MAX_VALUE) + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java:64]: @aloubyansky I find this pretty confusing, do the parenthesis `(Integer.MAX_VALUE)` have any effect? I am not sure where the cast will have effect, could you make it a bit clearer with more/less parenthesis?

Comment (aloubyansky) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java]: Yes, it's casting Integer.MAX_VALUE to a double before adding 1.

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java [hunk 2]
```diff
@@ -62,7 +61,7 @@ private double calculateScore(OriginCombination s) {
         for (OriginWithPreference o : s.getCollectedOrigins()) {
             combinationScore += Math.pow(extOrigins.size(),
                     highestRegistryPreference + 1 - o.getPreference().registryPreference)
-                    * ((double) (Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
+                    * (((double) Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java:64]:
```suggestion
                    * ((((double) Integer.MAX_VALUE) + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE);
```

@aloubyansky I would extract this to a static method and make sure it works as expected with unit tests, this is a tricky computation.

**Comment by ia3andy** [comment]
@aloubyansky I added a suggestion of testing, in the mean time I am building locally to try it

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java]: talking about this block 
```java
Math.pow(extOrigins.size(),
                    highestRegistryPreference + 1 - o.getPreference().registryPreference)
                    * (((double) Integer.MAX_VALUE + 1 - o.getPreference().platformPreference) / Integer.MAX_VALUE)
```

Comment (aloubyansky) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java]: The formula itself isn't particularly interesting to test in isolation. What counts is which platform and extension versions are suggested in different contexts, for that I do add tests.

Comment (ia3andy) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/selection/OriginSelector.java]: any formula is interesting to test in isolation IMHO to make sur the calculation is done right. Even more when there is casts and different types. For example, if registryPreference or platformPreference type changes, that could make it fail again, while with a strong typed test on
`computeScore(int, int, int, int)` (or whatever) make sure it is used right and valid in common cases.

This is just a suggestion.. I am not blocking the PR ;)

**Review by ia3andy — APPROVED** [review]
code.quarkus failing test is now passing

**Pull request merged**